packages/widget/src/components/common/buttons/connect-wallet.ts

```diff
@@ -62,49 +60,52 @@ export class ConnectWalletButton extends BaseComponent {
     return !!this.wallets.evmWallet || !!this.wallets.substrateWallet;
   }
 
-  render(): HTMLTemplateResult {
+  private renderConnectWalletButton(): HTMLTemplateResult | undefined {
+    if (this.wallets.substrateWallet) return;
+
+    return when(
+      this.isWalletConnected(),
+      () =>
+        html` <button
+          role="button"
+          @click=${this.onDisconnectClicked}
+          class="connectWalletButton"
+        >
+          Disconnect
+        </button>`,
+      () =>
+        html` <button
+          @click=${this.onConnectClicked}
+          role="button"
+          class="connectWalletButton"
+        >
+          ${plusIcon} Connect Wallet
+        </button>`
+    );
+  }
+
+  private renderWalletAddress(): TemplateResult {
     const evmWallet = this.wallets.evmWallet;
-    const substrateWallet = this.wallets.substrateWallet;
-    //TODO: this is wrong we need to enable user to select account
-    const substrateAccount = substrateWallet?.accounts[0];
+
+    return when(
```

Comment:
in case we add more network types in the future, lets use `choose` instead (which will also check connected substrate wallet)

Reply:
updated